forgerock-auth/src/main/java/org/forgerock/android/auth/callback/DeviceBindingCallback.kt

```diff
@@ -139,83 +162,112 @@ open class DeviceBindingCallback: AbstractCallback {
      *
      * @param context  The Application Context
      * @param listener The Listener to listen for the result
-     * @param authInterface Interface to find the Authentication Type
+     * @param deviceAuthenticator Interface to find the Authentication Type
      * @param encryptedPreference Persist the values in encrypted shared preference
      */
+    @OptIn(ExperimentalTime::class)
     @JvmOverloads
     internal fun execute(context: Context,
-                               listener: FRListener<Void>,
-                               authInterface: DeviceAuthenticator = getDeviceBindAuthenticator(),
-                               encryptedPreference: DeviceRepository = SharedPreferencesDeviceRepository(context),
-                               deviceId: String = DeviceIdentifier.builder().context(context).build().identifier) {
+                         listener: FRListener<Void>,
+                         deviceAuthenticator: DeviceAuthenticator = getDeviceAuthenticator(
+                             deviceBindingAuthenticationType),
+                         encryptedPreference: DeviceRepository = SharedPreferencesDeviceRepository(
+                             context),
+                         deviceId: String = DeviceIdentifier.builder().context(context)
+                             .build().identifier) {
+
+
+        deviceAuthenticator.initialize(userId, Prompt(title, subtitle, description))
 
-        if(authInterface.isSupported().not()) {
-            handleException(Unsupported(), listener = listener)
+        if (deviceAuthenticator.isSupported(context).not()) {
+            handleException(Unsupported(), e = null, listener = listener)
             return
         }
 
-        try {
-            val keypair = authInterface.generateKeys()
-            authInterface.authenticate(timeout ?: 60) { result ->
-                if (result is Success) {
-                    val kid = encryptedPreference.persist(userId, userName, keypair.keyAlias, deviceBindingAuthenticationType)
-                    val jws = authInterface.sign(keypair, kid, userId, challenge, getExpiration())
+        val scope = CoroutineScope(Dispatchers.Default)
```

Comment:
my understanding of Coroutines dispatcher is any 
 1. UI related logic should be part of Dispatchers.Main thread 
 2. Dispatchers.IO for networking calls, 
 3. Dispatchers.Default is for CPU intensive task

So we are displaying biometric prompt and selecting users keys should be part of Dispatchers.Main and other logics like generating keys should be part of Dispatcher.Default? can you confirm this.

Reply:
I moved UI request to Main thread, but an extra dependency library is required.